### PR TITLE
PPF-562 Create helper method to activate uitpas integration with organizers

### PR DESCRIPTION
**Pull request by grubolsch**
### Added
- Added new unit tests for UdbOrganizerRepository, class was untested.
- Added new check in EloquentIntegrationRepositoryTest to make sure Organizers are created in db.
- Added new Collection for UdbOrganizer
- 
### Changed
- Added 2 versions of UdbOrganizerRepository::create(), the new one is called "createInBulk", the later one wraps all inserts in a transaction.
- requestActivation and activateWithOrganization now wrap the creation of the Organizer.

---
Ticket: https://jira.uitdatabank.be/browse/PPF-562

### CODE REVIEW

File: app/Domain/Integrations/UdbOrganizerCollection.php [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+declare(strict_types=1);
+
+namespace App\Domain\Integrations;
+
+use Illuminate\Support\Collection;
+
+/**
+ * @extends Collection<int, UdbOrganizer>
+ */
+final class UdbOrganizerCollection extends Collection
```
Comment (LucWollants) [inline app/Domain/Integrations/UdbOrganizerCollection.php:12]: For other collections we used a naming convention to just use the plural form. For example `Realm` and `Realms`, so for this case it should be `UdbOrganizers`

File: app/Domain/Integrations/Repositories/EloquentIntegrationRepository.php [hunk 2]
```diff
@@ -115,15 +129,25 @@ public function activate(UuidInterface $id): void
         $integrationModel->activate();
     }
 
-    public function activateWithOrganization(UuidInterface $id, UuidInterface $organizationId, ?string $couponCode): void
+    public function activateWithOrganization(UuidInterface $integrationId, UuidInterface $organizationId, ?string $couponCode, array $organizers=[]): void
```
Comment (LucWollants) [inline app/Domain/Integrations/Repositories/EloquentIntegrationRepository.php:132]: Should this also be typed on the new collection?

File: app/Domain/Integrations/Repositories/EloquentUdbOrganizerRepository.php [hunk 3]
```diff
@@ -6,11 +6,21 @@
...
+    public function create(UdbOrganizer $organizer): void
+    {
+        UdbOrganizerModel::query()->create([
+            'id' => $organizer->id->toString(),
+            'integration_id' => $organizer->integrationId->toString(),
+            'organizer_id' => $organizer->organizerId,
+        ]);
+    }
+
+    public function createInBulk(UdbOrganizerCollection $organizers): void
```
Comment (LucWollants) [inline app/Domain/Integrations/Repositories/EloquentUdbOrganizerRepository.php:23]: You can reuse the `create` method inside the `createInBulk` method

File: app/Domain/Integrations/Repositories/IntegrationRepository.php [hunk 4]
```diff
@@ -17,8 +18,8 @@ public function getById(UuidInterface $id): Integration;
     public function getByIdWithTrashed(UuidInterface $id): Integration;
     public function deleteById(UuidInterface $id): ?bool;
     public function getByContactEmail(string $email, ?string $searchQuery): PaginatedCollection;
-    public function requestActivation(UuidInterface $id, UuidInterface $organizationId, ?string $couponCode): void;
+    public function requestActivation(UuidInterface $id, UuidInterface $organizationId, ?string $couponCode, UdbOrganizerCollection $organizers=null): void;
     public function activate(UuidInterface $id): void;
-    public function activateWithOrganization(UuidInterface $id, UuidInterface $organizationId, ?string $couponCode): void;
+    public function activateWithOrganization(UuidInterface $integrationId, UuidInterface $organizationId, ?string $couponCode, array $organizers=[]): void;
```
Comment (LucWollants) [inline app/Domain/Integrations/Repositories/IntegrationRepository.php:23]: Why change a single parameter? Is it not better to keep it consistent with the rest of the methods?

File: app/Domain/Integrations/Repositories/EloquentIntegrationRepository.php (hunk 2 above)
Comment (grubolsch) [reply app/Domain/Integrations/Repositories/EloquentIntegrationRepository.php:132]: I thought so to at first but if you look at ActivateIntegration you can see that actually it is an array of strings (ids).
I will add a docblock to make this more clear.

**Comment by grubolsch** [comment]
Feedback has been processed

File: app/Domain/Integrations/Controllers/IntegrationController.php [hunk 5]
```diff
@@ -329,10 +330,12 @@ public function requestActivation(string $id, RequestActivationRequest $request)
...
 
-        $organizers = UdbOrganizerMapper::mapActivationRequest($request, $id);
-        $this->organizerRepository->create(...$organizers);
-
-        $this->integrationRepository->requestActivation(Uuid::fromString($id), $organization->id, $request->input('coupon'));
+        $this->integrationRepository->requestActivation(
+            Uuid::fromString($id),
+            $organization->id,
+            $request->input('coupon'),
+            new UdbOrganizers(...UdbOrganizerMapper::mapActivationRequest($request, $id))
```
Comment (LucWollants) [inline app/Domain/Integrations/Controllers/IntegrationController.php:337]: Is it possible to make `mapActivationRequest` return `UdbOrganizers`? (I forgot to press "add comment" for this one)

File: app/Domain/Integrations/Repositories/EloquentIntegrationRepository.php (hunk 2 above)
Comment (LucWollants) [reply app/Domain/Integrations/Repositories/EloquentIntegrationRepository.php]: But should it be an array of strings? The `UdbOrganizers` could be created where this gets called. So in the `handle` method of `ActivateUitpasIntegration` the construction of `UdbOrganizer` could be done.

**Review by LucWollants — APPROVED** [review]

File: app/Domain/Integrations/Controllers/IntegrationController.php (hunk 5 above)
Comment (LucWollants) [reply app/Domain/Integrations/Controllers/IntegrationController.php]: I think it returns a single organization, so there is no need for a collection here.

File: app/Domain/Integrations/Repositories/EloquentIntegrationRepository.php (hunk 2 above)
Comment (grubolsch) [reply app/Domain/Integrations/Repositories/EloquentIntegrationRepository.php]: Moved it to the ActivateUitpasIntegration class

**Pull request merged**